drivers/entropy/entropy_stm32.c

```diff
@@ -381,7 +411,31 @@ static int entropy_stm32_rng_init(const struct device *dev)
 	 *  choose PLLSAI1 source as the 48 MHz clock is needed for the RNG
 	 *  Linear Feedback Shift Register
 	 */
-	 LL_RCC_SetRNGClockSource(LL_RCC_RNG_CLKSOURCE_PLLSAI1);
+	LL_RCC_SetRNGClockSource(LL_RCC_RNG_CLKSOURCE_PLLSAI1);
+
+	return 0;
+}
+#endif /* CONFIG_SOC_SERIES_STM32L4X */
+
+static int entropy_stm32_rng_init(const struct device *dev)
+{
+	struct entropy_stm32_rng_dev_data *dev_data;
+	const struct entropy_stm32_rng_dev_cfg *dev_cfg;
+	int res;
+
+	__ASSERT_NO_MSG(dev != NULL);
+
+	dev_data = DEV_DATA(dev);
+	dev_cfg = DEV_CFG(dev);
+
+	__ASSERT_NO_MSG(dev_data != NULL);
+	__ASSERT_NO_MSG(dev_cfg != NULL);
+
+#if CONFIG_SOC_SERIES_STM32L4X
+	res = entropy_stm32_rng_init_l4();
```

Comment:
Why is a L4 specific init required?

Reply:
Not my decision, I've just extended it. There was already a L4 specific code in entropy_stm32_rng_init().

Reply:
I know this part of code would require some update, but I prefer to see evolution in the sense of factorization and commonalisation rather than splitting out per series.

Reply:
Same observation as above.